django_project/frontend/views/users.py

```diff
@@ -0,0 +1,310 @@
+        if exists:
+            try:
+                invitation = OrganisationInvites.objects.get(
+                    email=email, organisation_id=organisation)
+                return invitation.joined
+            except OrganisationInvites.DoesNotExist:
+                return None
+        else:
+            return False
+
+    def calculate_rows_per_page(self, data):
+        total_rows = len(data)
+
+        desired_rows_per_page = 5
+
+        # Calculate the dynamic number of rows per page
+        rows_per_page = min(desired_rows_per_page, total_rows)
+
+        return rows_per_page
+
+
+    def search_table(self, request):
+        query = request.POST.get('query')
+        extracted_string = self.extract_substring(query)
+        # print(extracted_string)
+        matching_users = self.search_users(extracted_string)
+        # print(matching_users)
+
+        data = []
+
+        # search user within the orginisation
+        for user in matching_users:
+            try:
+                org_user = OrganisationUser.objects.get(user=user)
+                role = UserProfile.objects.get(user=user).user_role_type_id
+                role = str(role)
+            except OrganisationUser.DoesNotExist:
+                continue
+            except UserProfile.DoesNotExist:
+                role = None
+            if org_user:
+                if not org_user.user == self.request.user:
+                    data.append({
+                        'organisation': str(org_user.organisation),
+                        'user': str(org_user.user),
+                        'id': org_user.user.id,
+                        'role': role
+                        # Add more columns as needed
+                    })
+
+        return JsonResponse({'data': json.dumps(data, cls=DjangoJSONEncoder)})
+
+    def post(self, request):
+        # Default post method logic
+        return JsonResponse({'status': 'success'})
+
+    def invite_post(self, request):
+
+        # retrieve data from front end/html template
+        email = request.POST.get('email')
+        role = request.POST.get('inviteAs')
+        permissions = request.POST.get('memberRole')
+
+        # assign role from the roles defined in the db
+        if role == 'manager':
+            if permissions == 'write':
+                role = 'Admin'
+            else:
+                role = 'Base user'
+        else:
+            if permissions == 'write':
+                role = 'Admin'
+            else:
+                role = 'Base user'
+
+        # get role by name
+        user_role = self.get_user_role(role)
+
+        try:
+            # add invitation to model
+            is_new_invitation = self.is_new_invitation(
+                email, self.request.session[CURRENT_ORGANISATION_ID_KEY])
+            org_id = self.request.session[CURRENT_ORGANISATION_ID_KEY]
+            if not is_new_invitation:
+                if role == 'manager':
```

Comment:
I think you will not have role == 'manager' value here, because it has been overwritten with either Admin or Base user. You will need to fix role assignment at line 145.